src/nnet3/nnet-test-utils.cc

```diff
@@ -804,6 +804,24 @@ static void GenerateRandomComponentConfig(std::string *component_type,
       os << "input-dim=" << input_dim << " output-dim=" << output_dim;
       break;
     }
+    case 17: {
+      *component_type = "Convolutional1dComponent";
+      int32 patch_stride = 10 + Rand() % 50, patch_step = 1 + Rand() % 4, 
+	    patch_dim = 4 + Rand () % 5;
+
+      // decrease patch_stride until (patch_stride - patch_dim) % patch_step == 0
+      while ((patch_stride - patch_dim) % patch_step != 0)
```

Comment:
You can add that to this pull request.

On Sun, Sep 20, 2015 at 7:58 PM, Daniel Povey <dpovey@gmail.com> wrote:

> See GenerateConfigSequence and the functions it calls.
>
>
>
> On Sun, Sep 20, 2015 at 6:59 PM, Yiming Wang <notifications@github.com>
> wrote:
>
>> In src/nnet3/nnet-test-utils.cc
>> <https://github.com/kaldi-asr/kaldi/pull/156#discussion_r39936061>:
>>
>> > @@ -804,6 +804,24 @@ static void GenerateRandomComponentConfig(std::string *component_type,
>> >        os << "input-dim=" << input_dim << " output-dim=" << output_dim;
>> >        break;
>> >      }
>> > +    case 17: {
>> > +      *component_type = "Convolutional1dComponent";
>> > +      int32 patch_stride = 10 + Rand() % 50, patch_step = 1 + Rand() % 4,
>> > +	    patch_dim = 4 + Rand () % 5;
>> > +
>> > +      // decrease patch_stride until (patch_stride - patch_dim) % patch_step == 0
>> > +      while ((patch_stride - patch_dim) % patch_step != 0)
>>
>> I fixed that. Now there is only one commit of mine.
>> By a test example, did you mean a NN with Convolutional1dComponent,
>> rather than only testing a component? Could you please refer me to a
>> similar example in nnet-test-utils.cc or somewhere else?
>> Thanks,
>> Yiming
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/kaldi-asr/kaldi/pull/156/files#r39936061>.
>>
>
>